optimum/exporters/onnx/convert.py

```diff
@@ -232,7 +238,8 @@ def validate_model_outputs(
         reference_model.to(device)
 
         for key, value in reference_model_inputs.items():
-            reference_model_inputs[key] = recursive_to_device(value=value, device=device)
+            reference_model_inputs[key] = recursive_to_dtype(value=value, dtype=dtype)
+            reference_model_inputs[key] = recursive_to_device(value=reference_model_inputs[key], device=device)
```

Comment:
Nit: since both `recursive_to_dtype` and `recursive_to_device` ultimately rely on `tensor.Tensor.to`, we could actually have a unified `recursive_to_device_dtype` method.

Reply:
`recursive_to` applying to to a nested structure.
It could even be as simple as that:

```python
from torch.utils._pytree import tree_map

def recursive_to(possibly_nested_structure, *args, **kwargs):
     def map_fn(value):
          if isinstance(value, tensor):
              return value.to(*args, **kwargs)
         return value
     return tree_map(map_fn, possibly_nested_structure)
```

This function could be apply to nested structures of tensors.

Reply:
Thanks for the suggestion, but I'm not super keen on using undocumented private method from pytorch.